radix-engine/src/system/system_modules/costing/costing_entry.rs

```diff
@@ -290,6 +295,9 @@ pub mod owned {
         RefCheck {
             event: RefCheckEventOwned,
         },
+        Nullification {
+            io_access: Vec<IOAccess>,
```

Comment:
Naively, I'd have expected this to be split between execution cost for the read, and finalization cost for the write?

Reply:
Also I wonder if we should use more intuitive names here, for the purpose of the receipt? (because "Nullification" is not really a common name in crypto/literature, even if we're using it internally)
* `ExecutionCostingEntry::CheckIntentValidity`
* `FinalizationCostingEntry::RecordIntentCommits`

Reply:
Sounds reasonable to me.

Reply:
I've tried to add the whole deferred finalization cost, and then realized it doesn't fit the finalization cost model. 

Finalization costs are supported to reflect on post-execution costs in Node.

But here, we really want to just bill for execution time as a result of IOAccess. Intent hashes are periodically removed, so little overhead to the state storage.

Reply:
`FinalizationCostingEntry::RecordIntentCommits` is actually  covered by `FinalizationCostingEntry::CommitStateUpdates`

Reply:
Are you sure? When I looked at the code, finalisation of the state updates was calculated before the intent state changes were written

Reply:
You're right! I don't know what I was thinking. Let me add it explicitly. 

And looks like we don't need to defer this cost, but apply at finalization time. If it fails, there is no commit anyway. Even fairer to users.

Reply:
Sounds good.

And if we add billing for non-contingent intents (currently just transaction intents), these should possibly be billed up front/deferred.